AdaptorsFunctionality/test/unit/src/au/gov/asd/tac/constellation/functionality/adaptors/dataaccess/plugins/importing/ImportFromRDFPluginNGTest.java

```diff
@@ -51,46 +75,80 @@ public void setUpMethod() throws Exception {
     public void tearDownMethod() throws Exception {
     }
 
-    /**
-     * Test of query method, of class ImportFromRDFPlugin.
-     */
+
     @Test
-    public void testQuery() throws Exception {
+    public void testMusicTurtleQuery() throws Exception {
         final RecordStore query = new GraphRecordStore();
-        PluginInteraction interaction = null;
-        PluginParameters parameters = new PluginParameters();
-        ImportFromRDFPlugin instance = new ImportFromRDFPlugin();
-        RecordStore expResult = new GraphRecordStore();
-        RecordStore result = instance.query(query, interaction, parameters);
-        assertEquals(expResult, result);
+        final PluginInteraction interaction = null;
+        final ImportFromRDFPlugin instance = new ImportFromRDFPlugin();
+        final PluginParameters parameters = instance.createParameters();
+        parameters.setStringValue(ImportFromRDFPlugin.INPUT_FILE_URI_PARAMETER_ID, "file://" + this.getClass().getResource("./resources/music.ttl").getFile());
+
+        final RecordStore expResult = new GraphRecordStore();
+        final RecordStore result = instance.query(query, interaction, parameters);
+//        System.out.println(result.toStringVerbose());
+
+        final Set<String> expectedTypes = new TreeSet(result.getAll(GraphRecordStoreUtilities.SOURCE + AnalyticConcept.VertexAttribute.TYPE));
+        Set<String> actualTypes = new TreeSet<>();
+        actualTypes.add("Album");
+        actualTypes.add("Artist");
+        actualTypes.add("Band");
+        actualTypes.add("Song");
+        assertEquals(expectedTypes, actualTypes);
+
+//        assertEquals(expResult, result);
     }
 
-//    /**
-//     * Test of getType method, of class ImportFromRDFPlugin.
-//     */
-//    @Test
-//    public void testGetType() {
-//        System.out.println("getType");
-//        ImportFromRDFPlugin instance = new ImportFromRDFPlugin();
-//        String expResult = "";
-//        String result = instance.getType();
-//        assertEquals(result, expResult);
-//        // TODO review the generated test code and remove the default call to fail.
-//        fail("The test case is a prototype.");
-//    }
-//
-//    /**
-//     * Test of getPosition method, of class ImportFromRDFPlugin.
-//     */
+    /**
+     * Learning how to use the RDF4J Model class
+     *
+     * @throws Exception
+     */
 //    @Test
-//    public void testGetPosition() {
-//        System.out.println("getPosition");
-//        ImportFromRDFPlugin instance = new ImportFromRDFPlugin();
-//        int expResult = 0;
-//        int result = instance.getPosition();
-//        assertEquals(result, expResult);
-//        // TODO review the generated test code and remove the default call to fail.
-//        fail("The test case is a prototype.");
-//    }
+    public void testRdf4jModel() throws Exception {
```

Comment:
Adding this temporarily for testing how the RDF4J Model API works, can remove this method later. Doing stuff here means you don't need to run Constellation to test things.

Reply:
Agree, but in this initial stage, I reckon it's double work as the results can vary, requiring more work in the unit tests as well.